core/io/http_client.cpp

```diff
@@ -356,6 +356,12 @@ Error HTTPClient::poll() {
 					response_headers.clear();
 					response_num = RESPONSE_OK;
 
+					// Per the HTTP 1.1 spec, keep-alive is the default, but in practice
+					// it's safe to assume it only if the explicit header is found, allowing
+					// to handle body-up-to-EOF responses on naive servers; that's what Curl
+					// and browsers do
+					bool keep_alive = false;
```

Comment:
I would really rather make `keep_alive` an instance variable like `body_size` instead of using `body_size = -1` to keep track of `!keep_alive`

Reply:
I've ended up naming the instance variable `body_size_known`, so in this response setup part, the one reading the code can see it's a function of `body_size`, `chunked` and `keep_alive`, even if he doesn't read the comment.

Reply:
You're right. Maybe by just giving a different name to the variable all makes sense. `read_until_eof` would convey the exact meaning.